### PR TITLE
Fix binding project LinkWithAttributes generation to prevent unnecess…

### PR DESCRIPTION
**Pull request by chamons**
…ary rebuilds
- https://bugzilla.xamarin.com/show_bug.cgi?id=45762

### CODE REVIEW

**Comment by chamons** [comment]
Repeat of https://github.com/xamarin/xamarin-macios/pull/1017 but this time not early merged.

**Comment by chamons** [comment]
Similar in spirit to #1014 but should fix both XI and XM since logic is shared in this case.

This patch assumes PR 1014 lands first, It might not build w\o the other test changes (I just cherry picked for review).

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/PrepareNativeReferencesTaskBase.cs [hunk 1]
```diff
@@ -137,9 +137,16 @@ public override bool Execute ()
...
 			var linkWith = Path.Combine (IntermediateOutputPath, "LinkWithAttributes.cs");
-			Directory.CreateDirectory (Path.GetDirectoryName (linkWith));
-			File.WriteAllText (linkWith, text.ToString ());
+			if (File.Exists (linkWith)) {
+				string existingLinkWithText = File.ReadAllText (linkWith);
+				skipLinkWithGeneration = String.Equals (existingLinkWithText, text.ToString (), StringComparison.Ordinal);
+			}
+			if (!skipLinkWithGeneration) {
+				Directory.CreateDirectory (Path.GetDirectoryName (linkWith));
+				File.WriteAllText (linkWith, text.ToString ());
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/PrepareNativeReferencesTaskBase.cs]: you might be calling `text.ToString()` twice (not sure if it's cached, assuming it's a `StringBuilder`)

Comment (chamons) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/PrepareNativeReferencesTaskBase.cs]: @spouliot fixed.

**Comment by chamons** [comment]
@rolfbjarne Fixed. Will wait for build.

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

a) introspection-mac/Unified: random crash filed as bug [#46102](https://bugzilla.xamarin.com/show_bug.cgi?id=46102)
b) introspection-ios/watchOS: ibtool hung during build - filed as bug [#46097](https://bugzilla.xamarin.com/show_bug.cgi?id=46097)

So this is green from jenkins.

**Comment by rolfbjarne** [comment]
This is dependent on PR #1014, which is already merged.

**Pull request merged**